glib-macros/src/gboxed_shared_derive.rs

```diff
@@ -178,5 +178,40 @@ pub fn impl_gshared_boxed(input: &syn::DeriveInput) -> proc_macro2::TokenStream
         #impl_to_value_optional
 
         #impl_from_value
+
+        impl #crate_ident::translate::FromGlibPtrNone<*const #refcounted_type_prefix::InnerType> for #name {
+            #[inline]
+            unsafe fn from_glib_none(ptr: *const #refcounted_type_prefix::InnerType) -> Self {
+                let ptr = #refcounted_type_prefix::ref_(ptr);
+                #name(#refcounted_type_prefix::from_raw(ptr))
+            }
+        }
+
+        impl #crate_ident::translate::FromGlibPtrFull<*mut #refcounted_type_prefix::InnerType> for #name {
+            #[inline]
+            unsafe fn from_glib_full(ptr: *mut #refcounted_type_prefix::InnerType) -> Self {
+                #name(#refcounted_type_prefix::from_raw(ptr))
+            }
+        }
+
+        impl<'a> #crate_ident::translate::ToGlibPtr<'a, *const #refcounted_type_prefix::InnerType> for #name {
+            type Storage = &'a Self;
+
+            #[inline]
+            fn to_glib_none(&'a self) -> #crate_ident::translate::Stash<'a, *const #refcounted_type_prefix::InnerType, Self> {
+                let r = <#name as #crate_ident::subclass::shared::SharedType>::into_refcounted(self.clone());
+                unsafe {
+                    #crate_ident::translate::Stash(#refcounted_type_prefix::into_raw(r), self)
+                }
+            }
+
+            #[inline]
+            fn to_glib_full(&self) -> *const #refcounted_type_prefix::InnerType {
+                let r = <#name as #crate_ident::subclass::shared::SharedType>::into_refcounted(self.clone());
+                unsafe {
+                    #refcounted_type_prefix::into_raw(r)
```

Comment:
```suggestion
                    #refcounted_type_prefix::ref_(#refcounted_type_prefix::into_raw(r))
```

Maybe? Not entirely sure, either here that or in `to_glib_none()` it shouldn't use `into_raw()`. The way how it is now it is asymmetric.

Reply:
And this one has to be like in my suggestion

Reply:
as discussed in chat, we are cloning in the line above